banzai/tests/test_pattern_noise_qc.py

```diff
@@ -83,3 +86,56 @@ def test_pattern_noise_in_only_one_quadrant(set_random_seed):
     detector.logger.error = mock.MagicMock()
     detector.do_stage([image])
     assert detector.logger.error.called
+
+
+def test_trim_edges():
+    assert pattern_noise.trim_image_edges(np.zeros((100, 100)), fractional_edge_width=0.25).shape == (50, 50)
+    assert pattern_noise.trim_image_edges(np.zeros((100, 100)), fractional_edge_width=0.10).shape == (80, 80)
+    assert pattern_noise.trim_image_edges(np.zeros((100, 120)), fractional_edge_width=0.25).shape == (44, 64)
+
+
+def test_get_2d_power_band():
+    data = np.random.normal(0.0, 10.0, size=(100, 100))
+    fft = abs(np.fft.rfft2(data))[37:62]
+    power_band = pattern_noise.get_2d_power_band(data, fractional_band_width=0.25)
+    assert power_band.shape == (25, 51)
+    np.testing.assert_allclose(power_band, fft)
+
+
+def test_compute_snr():
+    data = np.random.normal(1000.0, 20.0, size=(200, 100))
+    snr = pattern_noise.compute_snr(data)
+    assert len(snr) == data.shape[1]-1
+    assert all(snr < 5)
+
+
+def test_get_odd_integer():
+    assert pattern_noise.get_odd_integer(1.5) == 3
```

Comment:
Shouldn't this go to 1?

Reply:
i.e. the closest odd integer to 1.5 is 1.

Reply:
I guess this function really finds the odd integer ceiling. I have updated the docstring, I can also modify the function to really find the closest odd integer but I think that may be overkill.